packages/ingest/granule.js

```diff
@@ -574,6 +578,131 @@ function selector(type, protocol) {
+  }
+}
+
+/**
+ * updates cmrFile metadata for any files being moved
+ *
+ * @param {Object} cmrFile - cmrFile to be updated
+ * @param {Object[]} sourceFiles - array of file objects
+ * @param {Object[]} destinations - array of objects defining the destination of granule files
+ * @param {string} distEndpoint - distribution enpoint from config
+ * @returns {Promise<undefined>} returns `undefined` when all the files are moved
+ */
+async function updateMetadata(cmrFile, sourceFiles, destinations, distEndpoint) {
+  const urls = [];
+  const file = cmrFile.file;
+  const destination = cmrFile.destination;
+
+  if (!buckets) {
+    const bucketsString = await aws.s3().getObject({
+      Bucket: process.env.bucket,
+      Key: `${process.env.stackName}/workflows/buckets.json`
+    }).promise();
+    buckets = JSON.parse(bucketsString.Body);
+  }
+
+  const bucketKeys = Object.keys(buckets);
+  sourceFiles.forEach((sourceFile) => {
+    const urlObj = {};
+    const currDestination = destinations.find((dest) => sourceFile.name.match(dest.regex));
+    let key;
+    let filepath;
+    if (currDestination) {
+      key =
+        bucketKeys.find((bucketKey) => currDestination.bucket.match(buckets[bucketKey].name));
+      filepath = currDestination.filepath;
+    }
+    else {
+      key = bucketKeys.find((bucketKey) => sourceFile.bucket.match(buckets[bucketKey].name));
+      filepath = sourceFile.filepath;
+    }
+    if (buckets[key].type.match('protected')) {
+      const extension = urljoin(buckets[key].name, `${filepath}/${sourceFile.name}`);
+      urlObj.URL = urljoin(distEndpoint, extension);
+      urlObj.URLDescription = 'File to download';
+      urls.push(urlObj);
+    }
+    else if (buckets[key].type.match('public')) {
+      urlObj.URL = `https://${buckets[key].name}.s3.amazonaws.com/${filepath}/${sourceFile.name}`;
+      urlObj.URLDescription = 'File to download';
+      urls.push(urlObj);
+    }
+  });
+  const metadata = await getMetadata(file.filename);
+  const metadataObject = await parseXmlString(metadata);
+  const metadataGranule = get(metadataObject, 'Granule');
+  const updatedGranule = {};
+  Object.keys(metadataGranule).forEach((key) => {
+    if (key === 'OnlineResources' || key === 'Orderable') {
+      updatedGranule.OnlineAccessURLs = {};
+    }
+    updatedGranule[key] = metadataGranule[key];
+  });
+  updatedGranule.OnlineAccessURLs.OnlineAccessURL = urls;
+  metadataObject.Granule = updatedGranule;
+  const builder = new xml2js.Builder();
+  const xml = builder.buildObject(metadataObject);
+  let action;
+  if (destination) {
+    const options = {
+      Bucket: file.bucket,
+      Key: file.filepath
+    };
+    const target = {
+      bucket: destination.bucket,
+      key: `${destination.filepath}/${file.name}`,
+      body: xml
+    };
+    action = postS3Object(target, options);
+  }
+  else {
+    action = postS3Object({ bucket: file.bucket, key: file.filepath, body: xml });
+  }
+  return action;
```

Comment:
the docstring says this function returns `undefined` but it seems to be actually returning something

Reply:
changed